core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/TermFilter.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2020 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.core.request;
+
+import static com.b2international.index.query.Expressions.scriptScore;
+
+import java.io.Serializable;
+import java.util.List;
+
+import com.b2international.index.query.Expression;
+import com.b2international.index.query.Expressions;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+
+/**
+ * Represents a {@link Serializable} encapsulation of term search configuration.
+ * 
+ * @since 7.11
+ */
+public class TermFilter implements Serializable {
+
+	private static final long serialVersionUID = 1L;
+	
+	private static final String PREFIX = ".pefix";
+	private static final String EXACT = ".exact";
+	private static final String ORIGINAL = ".original";
+	
+	private final String term;
+	private final Integer minShouldMatch;
```

Comment:
There are no validations present on the `minShouldMatch` field. It should be greater than or equal to `1`, otherwise throw a `BadRequestException`.